docs/DeveloperGuide.md

```diff
@@ -1,38 +1,275 @@
 # Developer Guide
+Welcome to the Meal Companion Developer Guide! Thank you for taking an interest in the behind-the-scenes working of our product, and we hope this document proves informative and useful for your work.
+
+## Table of Contents
+1. [Acknowledgements](#acknowledgements)
+2. [Setting up, getting started](#setting-up-getting-started)
+3. [Design](#design)
+   - [Command Parsing](#command-parsing)
+   - [Ingredient Class](#ingredient-class)
+   - [Recipe Class](#recipe-class)
+4. [Implementation](#implementation)
+   - [Add and Remove Command](#add-and-remove-command)
+   - [Recipe Detail Command](#recipe-detail-command)
+   - [Recipe Possible Command](#recipe-possible-command)
+   - [Recipe All Command](#recipe-all-command)
+   - [Storage Feature](#storage-feature)
+5. [Documentation, logging, testing, configuration, dev-ops](#documentation-logging-testing-configuration-dev-ops)
+6. [Appendix: Requirements](#appendix-requirements)
+   - [Product scope](#product-scope)
+   - [User stories](#user-stories)
+   - [Use cases](#use-cases)
+   - [Non-Functional Requirements](#non-functional-requirements)
+   - [Glossary](#glossary)
+7. [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
+
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+###### [Back to table of contents](#table-of-contents)
+
+## Setting up, getting started
```

Comment:
Very clear instructions, however there is generally no need to use the terminal in Intellij to test the functionality of the program!